src/config.rs

```diff
@@ -0,0 +1,1235 @@
+
+    /// The user account the daemon should run as, as either a username or a UID.
+    /// If the daemon cannot change to this UID on startup, it will exit.
+    /// If this element is not present, the daemon will not change or care about its UID.
+    pub user: Option<String>,
+}
+
+impl TryFrom<Document> for BusConfig {
+    type Error = Error;
+
+    fn try_from(value: Document) -> std::result::Result<Self, Self::Error> {
+        let mut bc = BusConfig::default();
+
+        for element in value.busconfig {
+            match element {
+                Element::AllowAnonymous => bc.allow_anonymous = true,
+                Element::Auth(s) => {
+                    bc.auth.insert(s);
+                }
+                Element::Fork => bc.fork = true,
+                Element::KeepUmask => bc.keep_umask = true,
+                Element::Limit => {
+                    eprintln!("warning: busd does not implement `<limit>`");
+                }
+                Element::Listen(s) => {
+                    bc.listen.insert(s);
+                }
+                Element::Pidfile(p) => bc.pidfile = Some(p),
+                Element::Policy(pe) => {
+                    if let Some(p) = OptionalPolicy::try_from(pe)? {
+                        bc.policies.push(p);
+                    }
+                }
+                Element::Servicedir(p) => {
+                    bc.servicedir.push(p);
+                }
+                Element::Servicehelper(p) => {
+                    // NOTE: we're assuming this has the same "last one wins" behaviour as `<type>`
+                    // TODO: warn and then ignore if we aren't reading:
+                    // /usr/share/dbus-1/system.conf
+                    bc.servicehelper = Some(p);
+                }
+                Element::StandardSessionServicedirs => {
+                    // NOTE: because servicedir sequence is important, we cannot de-duplicate this
+                    // TODO: warn and then ignore if we aren't reading: /etc/dbus-1/session.conf
+                    // TODO: include XDG_...-based directories
+                    bc.servicedir
+                        .push(PathBuf::from("/usr/share/dbus-1/services"));
+                }
+                Element::StandardSystemServicedirs => {
+                    // NOTE: because servicedir sequence is important, we cannot de-duplicate this
+                    // TODO: warn and then ignore if we aren't reading:
+                    // /usr/share/dbus-1/system.conf
+                    bc.servicedir
+                        .extend(STANDARD_SYSTEM_SERVICEDIRS.iter().map(PathBuf::from));
+                }
+                Element::Syslog => bc.syslog = true,
+                Element::Type(TypeElement { r#type: value }) => bc.r#type = Some(value),
+                Element::User(s) => bc.user = Some(s),
+            }
+        }
+
+        Ok(bc)
+    }
+}
+
+impl BusConfig {
+    pub fn limits(&self) -> Limits {
```

Comment:
If the limits are not coming from the config, we shouldn't them here. Limits can be their own mod/type independent of the config.